Всё переписано на Socket.io, добавлен дополнительный функционал

NSkye opened the pull request:
- всё переписано на socket.io
- исправлены минорные недочеты
- после потери соединения или провала его установки предпринимается ещё 10 попыток установить соединение
- после провала всех попыток установления соединения, страница перезагружается, чтоб вновь была доступна кнопка Join
- при успешном повторном переподключении после потери связи игра заново не инциализируется, а использует экземпляр, который был создан при первой инициализации
- кошка всё ещё на месте

Commit:
Переписано на socket.io

Commit:
Ограничение количества попыток реконнекта и рефакторинг

Commit:
Небольшие правки

Commit:
Исправлено "assigment to undeclared variable"

Commit:
Исправление опечаток

Commit:
Добавлено событие на случай провала всех попыток переподключения.

NikitaRudenko commented on main/index.js:
```diff
@@ -21,60 +21,84 @@
+    const type = message.type;
+    const data = message.data;
+    switch (type) {
+      case 'INITIALIZE':
```
тело каждого `case` лучше вынести в отдельную функцию, таким образом тестировать и поддерживать код станет в разы проще

Commit:
Рефакторинг

Commit:
Исправления по коду
- Код теперь лучше структурирован
- Для всех операций теперь используется класс LifeGameClient
- Конструктор класса принимает адрес(string), токен(string), стоит ли
производить повторные подключения(boolean) и количество попыток
повторного подключения(number)

NSkye committed:
Update index.js

NSkye committed:
Update index.js
